sdk/monitor/Azure.Monitor.Ingestion/tests/IngestionClientTest.cs

```diff
@@ -0,0 +1,227 @@
+        }
+        #endregion
+
+        [Test]
+        public async Task ValidInputFromObjectAsJson()
+        {
+            ClientSecretCredential clientSecretCrendential = new ClientSecretCredential("<tenantId>", "<cliendId>",  "<clientSecret>");
+            var dcrImmutableId = "<dcrImmutableId>";
+            var dcrEndpoint = "https://nibhati-dce-ku6s.westus2-1.ingest.monitor.azure.com";
+            var currentTime = DateTimeOffset.UtcNow.ToString("O");
+
+            BinaryData data = BinaryData.FromObjectAsJson(
+                // Use an anonymous type to create the payload
+                new[] {
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer1",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user1",
+                            TimeZone = "Pacific Time",
+                            Level = 4,
+                            CounterName = "AppMetric1",
+                            CounterValue = 15.3
+                        }
+                    },
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer2",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user2",
+                            TimeZone = "Central Time",
+                            Level = 3,
+                            CounterName = "AppMetric1",
+                            CounterValue = 23.5
+                        }
+                    },
+                });
+
+            LogsIngestionClient client = new(
+                new Uri(dcrEndpoint),
+                clientSecretCrendential);
+
+            // Make the request
+            Response response = client.Upload(dcrImmutableId, "Custom-MyTableRawData", RequestContent.Create(data)); //takes StreamName not tablename
+
+            // Check the response
+            Assert.AreEqual(204, response.Status);
+            Assert.AreEqual("", response.Content.ToString());
+
+            LogsQueryClient logsQueryClient = new LogsQueryClient(clientSecretCrendential);
+            var batch = new LogsBatchQuery();
+            string workspaceId = "22101c38-d67f-4ded-994e-67f093aff9f4";
+            string countQueryId = batch.AddWorkspaceQuery(
+                workspaceId,
+                "MyTable_CL | count;",
+                new QueryTimeRange(TimeSpan.FromDays(1)));
+
+            Response<LogsBatchQueryResultCollection> responseLogsQuery = await logsQueryClient.QueryBatchAsync(batch);
+
+            // Check the Azure.Monitor.Query Respose
+            Assert.AreEqual(200, responseLogsQuery.GetRawResponse().Status);
+            Assert.AreEqual("OK", responseLogsQuery.GetRawResponse().ReasonPhrase);
+            Assert.IsTrue(responseLogsQuery.Value.GetResult<int>(countQueryId).Single() >= 2);
```

Comment:
This is fine for now, but we should change these in the near future to inject a random value (using `Recoding.Random`) and then query for the presence of that value in the logs.  That'll make this more resilient to multiple people testing against the same resource (or you running the same tests a few times in a row).